pydantic_ai_slim/pydantic_ai/ui/web/api.py

```diff
@@ -0,0 +1,94 @@
+"""API router for the web chat UI."""
+
+from typing import Annotated
+
+from fastapi import APIRouter, Depends, Request, Response
+from pydantic import BaseModel
+from pydantic.alias_generators import to_camel
+
+from pydantic_ai import Agent
+from pydantic_ai.builtin_tools import BUILTIN_TOOL_ID
+from pydantic_ai.ui.vercel_ai._adapter import VercelAIAdapter
+
+from .agent_options import (
+    AIModel,
+    BuiltinToolDef,
+    builtin_tool_definitions as default_builtin_tool_definitions,
+    models as default_models,
+)
+
+
+def get_agent(request: Request) -> Agent:
+    """Get the agent from app state."""
+    agent = getattr(request.app.state, 'agent', None)
```

Comment:
Do we need this? I think e can pass the agent into `create_api_router` directly

Reply:
I'm not sure what you mean about this one, would appreciate we clear it up in a quick call